src/lucky_record/associations.cr

```diff
@@ -9,9 +9,20 @@ module LuckyRecord::Associations
 
   macro belongs_to(type_declaration)
     {% assoc_name = type_declaration.var }
-    {% model = type_declaration.type %}
 
-    field {{ assoc_name.id }}_id : Int32
+    {% if type_declaration.type.is_a?(Union) %}
+      {% model = type_declaration.type.types.first %}
+      {% nilable = true %}
+    {% else %}
+      {% model = type_declaration.type %}
+      {% nilable = false %}
+    {% end %}
+
+    {% if nilable %}
+      field {{ assoc_name.id }}_id : Int32?
+    {% else %}
+      field {{ assoc_name.id }}_id : Int32
+    {% end %}
```

Comment:
Instead of doing two separate lines you could use `if nilable` to show the `?`

```crystal
field {{ assoc_name.id }}_id : Int32{% if nilable %}?{% end %}
```

Reply:
Wow I didn't know I could do that